libraries/eosiolib/core/eosio/utility.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+namespace eosio {
+   /* utility function to swap the endianness of a variable */
+   template <typename T>
+   inline T swap_endian(T val) {
+      static_assert(sizeof(T) <= 16);
+      if constexpr (sizeof(T) == 1)
+         return val;
+      else if constexpr (sizeof(T) == 2)
+         return (val << 8) |(val >> 8);
+      else if constexpr (sizeof(T) == 4) {
+         union i32_swap {
```

Comment:
Using a union like this may have undefined behavior depending on T.  signed int may alias unsigned int, but not long even if int and long have the same size. Use memcpy instead.